### PR TITLE
Replace DimObjective with DestinyObjectiveProgress

### PR DESCRIPTION
**Pull request by bhollis**
This PR does two things:

1. It entirely replaces the custom DimObjective structure with the DimObjectiveProgress from the API. This lets us consolidate our rendering logic into a single Objective component.
2. Rework the objective display to handle some of the newer objective types, like times, multipliers, etc. This works wherever objectives are displayed.

### CODE REVIEW

File: src/app/inventory/store/d1-item-factory.ts [hunk 1]
```diff
@@ -472,21 +472,6 @@ function makeItem(
     createdItem.equippingLabel = undefined;
   }
 
-  // do specific things for specific items
-  if (createdItem.hash === 491180618) {
-    // Trials Cards
```
Comment (bhollis) [inline src/app/inventory/store/d1-item-factory.ts:477]: Yes, special rendering for Trials cards in D1 is a thing of the past.

File: src/app/inventory/store/d2-item-factory.ts [hunk 2]
```diff
@@ -482,15 +482,12 @@ export function makeItem(
 
   // Compute complete / completion percentage
   if (createdItem.objectives) {
-    // Counter objectives for the new emblems shouldn't count.
```
Comment (bhollis) [inline src/app/inventory/store/d2-item-factory.ts:485]: With stat trackers, this is no longer a thing.

File: src/app/progress/D2SupplementalManifestDefinitions.ts [hunk 3]
```diff
@@ -1,79 +0,0 @@
...
-      minimumVisibilityThreshold: Number.MAX_VALUE,
-      progressDescription: '',
-      valueStyle: DestinyUnlockValueUIStyle.Automatic,
-      displayProperties: {
-        hasIcon: false,
-        icon: ''
-      },
-      completionValue: 0
-    },
-    // Encrypted Cache Key
```
Comment (bhollis) [inline src/app/progress/D2SupplementalManifestDefinitions.ts:30]: I didn't realize these had been removed from the game.

File: src/app/inventory/store/d1-item-factory.ts (hunk 1 above)
Comment (delphiactual) [reply src/app/inventory/store/d1-item-factory.ts:477]: It does not matter as trials is not available in D1 anymore.

File: src/app/inventory/store/d1-item-factory.ts [hunk 4]
```diff
@@ -733,60 +718,6 @@ function buildTalentGrid(item, talentDefs, progressDefs): D1TalentGrid | null {
...
-  function buildObjective(
-    name: string,
-    current: number,
-    max: number,
-    bool: boolean,
-    style?: string
-  ): DimObjective {
-    // t('TrialsCard.FiveWins')
-    // t('TrialsCard.SevenWins')
-    // t('TrialsCard.Flawless')
```
Comment (delphiactual) [inline src/app/inventory/store/d1-item-factory.ts:748]: these can be removed from i18n.json and dim.json as well

File: src/app/dim-ui/RichDestinyText.tsx [hunk 5]
```diff
@@ -56,7 +56,7 @@ const baseConversionTable: {
 }[] = [
   // Damage Types
   { objectiveHash: 2178780271, unicode: '', substring: null, icon: dmgArc       },
-  { objectiveHash:  400711454, unicode: '', substring: null, icon: dmgVoid      },
+  { objectiveHash: 3286925104, unicode: '', substring: null, icon: dmgVoid      },
   { objectiveHash: 2994623161, unicode: '', substring: null, icon: dmgSolar     },
   { objectiveHash: 1554970245, unicode: '', substring: null, icon: dmgKinetic   },
   // Precision
```
Comment (delphiactual) [inline src/app/dim-ui/RichDestinyText.tsx:62]: we need to update headshot to hash `2258255959` as well. We might need to move this table generation to d2ai

Comment (bhollis) [reply src/app/dim-ui/RichDestinyText.tsx:62]: Yeah that’d be good. I just happened to notice this one.

**Pull request merged**